src/EditorFeatures/Core/Implementation/Formatting/FormatCommandHandler.cs

```diff
@@ -56,19 +57,31 @@ internal partial class FormatCommandHandler :
 
         private void Format(ITextView textView, Document document, TextSpan? selectionOpt, CancellationToken cancellationToken)
         {
-            var formattingService = document.GetRequiredLanguageService<IEditorFormattingService>();
+            var formattingService = document.GetLanguageService<IFormattingInteractionService>();
+            var editorFormattingService = formattingService == null ? document.GetRequiredLanguageService<IEditorFormattingService>() : null;
 
             using (Logger.LogBlock(FunctionId.CommandHandler_FormatCommand, KeyValueLogMessage.Create(LogType.UserAction, m => m["Span"] = selectionOpt?.Length ?? -1), cancellationToken))
             using (var transaction = CreateEditTransaction(textView, EditorFeaturesResources.Formatting))
             {
-                var changes = formattingService.GetFormattingChangesAsync(document, selectionOpt, cancellationToken).WaitAndGetResult(cancellationToken);
+                var changes = GetChanges();
                 if (changes.Count == 0)
                 {
                     return;
                 }
 
                 ApplyChanges(document, changes, selectionOpt, cancellationToken);
                 transaction.Complete();
+
+                IList<TextChange> GetChanges()
```

Comment:
```suggestion
                transaction.Complete();
                
                return;

                IList<TextChange> GetChanges()
```

Having an explicit return makes it trivial to know that the local function isn't between actual code that executes as part of the main body.